tests/app.test.js

```diff
@@ -355,6 +357,107 @@ describe('InVEST global settings: dialog interactions', () => {
     expect(input.classList.contains('is-invalid')).toBeTruthy();
     expect(getByText('Save Changes')).toBeDisabled();
   });
+
+  test('Save global settings', async () => {
+    const nWorkers = '2';
+    const loggingLevel = 'DEBUG';
+    const nWorkersLabelText = 'Taskgraph n_workers parameter';
+    const loggingLabelText = 'Logging threshold';
+
+    const { getByText, getByLabelText, getByTitle } = render(
+      <App investExe="foo" />
+    );
+
+    fireEvent.click(getByTitle('settings'));
+    const nWorkersInput = getByLabelText(nWorkersLabelText, { exact: false });
+    const loggingInput = getByLabelText(loggingLabelText, { exact: false });
+
+    // Test that the default values when no global-settings file exists are
+    // loaded. I've found this helps allow componentDidMount processes to
+    // finish
+    await waitFor(() => {
+      expect(nWorkersInput).toHaveValue("-1");
+      expect(loggingInput).toHaveValue("INFO");
+    });
+
+    // Change the value
+    fireEvent.change(nWorkersInput, { target: { value: nWorkers } });
+    fireEvent.change(loggingInput, { target: { value: loggingLevel } });
+    await waitFor(() => { // the value to test is inherited through props
+      expect(nWorkersInput).toHaveValue(nWorkers);
+      expect(loggingInput).toHaveValue(loggingLevel);
+    });
+    // The real test: values saved to global-settings
+    fireEvent.click(getByText('Save Changes'));
+    // Check values in the investSettingsStore
+    const investSettingsStore = testables.getSettingsStore();
+    expect(investSettingsStore.store.nWorkers).toBe('2');
+    expect(investSettingsStore.store.loggingLevel).toBe('DEBUG');
+  });
+
+  test('Load global settings', async () => {
+    const expectedSettings = {
+      nWorkers: '-1',
+      loggingLevel: 'INFO'
+    }
+    const nWorkersLabelText = 'Taskgraph n_workers parameter';
+    const loggingLabelText = 'Logging threshold';
+
+    const { getByText, getByLabelText, getByTitle } = render(
+      <App investExe="foo" />
+    );
+
+    fireEvent.click(getByTitle('settings'));
+    const nWorkersInput = getByLabelText(nWorkersLabelText, { exact: false });
+    const loggingInput = getByLabelText(loggingLabelText, { exact: false });
+
+    // Test that the global-settings were loaded in from store.
```

Comment:
Damn Dave you're good. Yes, I actually punted on this. I tried a bunch of things to `re-mock` `createInstance` to set different values, but I could not get it to work. Would you know how to set a one time mock for `createInstance` in a single test?

Reply:
@dcdenu4 yeah that does sound like a nightmare - mocking something that's already mocked and keeping track of which takes precedence, etc. Instead of that, what if the test setup somehow calls `investSettingsStore.setItem(...)`? There's probably a few ways to do that. 

One way could be designing a `Settings` object that handles interfacing with the localforage instance, like we have with `InvestJob`. So then we can use that interface in the app and in test setup. Though I'm not 100% sure how this would work if test setup triggers a mocked `createInstance` and then the app render triggers another mocked `createInstance`. The real `localforage` seems to handle that well and re-use the same instance, but not sure if the mock is setup to do that. `InvestJob.test.js` has some examples like this.
```
test('save method works with no pre-existing database', async () => {
    const job = new InvestJob({
      modelRunName: 'foo',
      modelHumanName: 'Foo',
    });
    job.setProperty('argsValues', baseArgsValues);
    const recentJobs = await job.save();
    expect(recentJobs[0]).toBe(job.metadata);
  });
```

Reply:
Or I guess it's not the app `render` that triggers `createInstance` but just the `app.jsx` import.